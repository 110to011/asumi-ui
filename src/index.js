/**
 * Created by Elly on 2016/5/27.
 */

import Menu from './component/menu';
import SubMenu from './component/menu/subMenu';
import MenuItem from './component/menu/menuItem';
import MenuItemGroup from './component/menu/menuItemGroup';
import Tag from './component/tag';
import Grid from './component/grid';
import Form from './component/form';
import Tabs from './component/tabs';
import Radio from './component/radio';
import Group from './component/group';
import Input from './component/input';
import Modal from './component/modal';
import Table from './component/table';
import Editor from './component/editor';
import Col from './component/table/col';
import Button from './component/button';
import Select from './component/select';
import Loading from './component/loading';
import Popover from './component/popover';
import Message from './component/message';
import Animate from './component/animate';
import Upload from './component/upload';
import Checkbox from './component/checkbox';
import Dropdown from './component/dropdown';
import Option from './component/select/option';
import Pagination from './component/pagination';
import FormItem from './component/form/formItem';
import TabPanel from './component/tabs/tabPanel';
import RadioGroup from './component/radio/radioGroup';
import CheckboxGroup from './component/checkbox/checkGroup';
import SimplePagination from './component/pagination/simplePagination';
import DateTime from './component/datetime';

const Tooltip = Popover;

Table.Col = Col;

Select.Option = Option;

Tabs.TabPanel = TabPanel;

Menu.SubMenu = SubMenu;
Menu.MenuItem = MenuItem;
Menu.MenuItemGroup = MenuItemGroup;

Radio.Group = RadioGroup;
Checkbox.Group = CheckboxGroup;

Form.FormItem = FormItem;

Pagination.Simple = SimplePagination;

export {
    Col,
    Tag,
    Tabs,
    Menu,
    Form,
    Grid,
    Table,
    Modal,
    Input,
    Group,
    Radio,
    Select,
    Button,
    Upload,
    Option,
    Editor,
    SubMenu,
    Animate,
    Loading,
    Popover,
    Message,
    Tooltip,
    FormItem,
    MenuItem,
    TabPanel,
    Dropdown,
    Checkbox,
    DateTime,
    Pagination,
    RadioGroup,
    CheckboxGroup,
    MenuItemGroup,
    SimplePagination,
};