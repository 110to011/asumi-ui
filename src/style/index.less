@charset "utf-8";

@import "variables";
@import "normalize";
@import "fontawesome/less/font-awesome";

body {
  font-family: '微软雅黑', "Microsoft YaHei", Arial, Microsoft YaHei, Helvetica, sans-serif;
  margin: 0;
  color: @normalText;
  font-size: @fontSize;
  line-height: 1.42857;
}

a {
  color: @primary;
  &:hover, &:active, &:visited, &:focus {
    color: @d_primary;
  }
}

.el-shadow {
  box-shadow: 0 6px 12px @shadow;
  -webkit-box-shadow: 0 6px 12px @shadow;
}

.overflow {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

@import "nav";
@import "tabs";
@import "drop";
@import "caret";
@import "modal";
@import "input";
@import "table";
@import "button";
@import "stretch";
@import "popover";
@import "checkbox";
@import "pagination";
