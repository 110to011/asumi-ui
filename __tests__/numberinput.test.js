import React, {Component} from 'react';
import {shallow, mount} from 'enzyme';
import {NumberInput} from '../src';

describe('NumberInput', () => {
    it('create component', () => {
        const component = mount(<NumberInput/>);
        expect(component).toMatchSnapshot();
    });
    it('define disabled', () => {
        const component = mount(<NumberInput disabled/>);
        expect(component).toMatchSnapshot();
    });
    it('define size[small]', () => {
        const component = mount(<NumberInput size="small"/>);
        expect(component).toMatchSnapshot();
    });
    it('define size[large]', () => {
        const component = mount(<NumberInput size="large"/>);
        expect(component).toMatchSnapshot();
    });
    it('define className', () => {
        const component = shallow(<NumberInput className="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('.jest').length).toEqual(1);
    });
    it('define style', () => {
        const component = shallow(<NumberInput style={{width: 100}}/>);
        expect(component).toBeDefined();
        expect(component.instance().props.style.width).toEqual(100);
    });
    it('define name', () => {
        const component = mount(<NumberInput name="jest"/>);
        expect(component).toBeDefined();
        expect(component.find('input[name="jest"]').length).toEqual(1);
    });
    it('define readOnly', () => {
        const component = mount(<NumberInput readOnly/>);
        expect(component).toBeDefined(); //喵喵？
        expect(component.find('.el-input input[readOnly]').length).toEqual(1);
    });
    it('define prepend', () => {
        const component = mount(<NumberInput prepend={<em/>}/>);
        expect(component).toBeDefined();
        expect(component.find('.el-input-wrapper .el-input-prepend em').length).toEqual(1);
    });
    it('define append', () => {
        const component = mount(<NumberInput append={<em/>}/>);
        expect(component).toBeDefined();
        expect(component.find('.el-input-wrapper .el-input-append em').length).toEqual(0);
    });
    it('define icon', () => {
        const component = mount(<NumberInput icon={<em/>}/>);
        expect(component).toBeDefined();
        expect(component.find('.el-input-wrapper .el-input-icon em').length).toEqual(1);
    });
    it('when click handler', () => {
        const component = mount(<NumberInput/>);
        component.find('.el-number-input-up').simulate('click');
        expect(component.instance().state.value).toEqual(1);
        component.find('.el-number-input-down').simulate('click');
        expect(component.instance().state.value).toEqual(0);
    });
    it('define step', () => {
        const component = mount(<NumberInput step={5}/>);
        component.find('.el-number-input-up').simulate('click');
        expect(component.instance().state.value).toEqual(5);
        component.find('.el-number-input-down').simulate('click');
        expect(component.instance().state.value).toEqual(0);
    });
    it('define min & max', () => {
        const component = mount(<NumberInput min={0} max={2}/>);
        component.find('.el-number-input-up').simulate('click');
        component.find('.el-number-input-up').simulate('click');
        component.find('.el-number-input-up').simulate('click');
        expect(component.instance().state.value).toEqual(2);
        component.find('.el-number-input-down').simulate('click');
        component.find('.el-number-input-down').simulate('click');
        component.find('.el-number-input-down').simulate('click');
        expect(component.instance().state.value).toEqual(0);
    });
    it('define dataFormat', () => {
        const component = mount(<NumberInput step={5} dataFormat={(data) => {
            return '+' + data
        }}/>);
        component.find('.el-number-input-up').simulate('click');
        expect(component.instance().state.value).toEqual(5);
        expect(component.instance().state.renderValue).toEqual("+5");
        component.find('.el-number-input-down').simulate('click');
        expect(component.instance().state.value).toEqual(0);
        expect(component.instance().state.renderValue).toEqual("+0");
    });
    it('input decimals', () => {
        const component = mount(<NumberInput/>);
        component.find('input').simulate('change', {target: {value: "1.1"}});
        expect(component.instance().state.value).toEqual("1.1");
        component.find('.el-number-input-down').simulate('click');
        expect(component.instance().state.value).toEqual(0.1);
    });
    it('onChange', () => {
        class Demo extends Component {
            constructor() {
                super();
                this.state = {
                    value: ""
                }
            }

            handleChange({value}) {
                this.setState({value});
            }

            render() {
                return <NumberInput onChange={this.handleChange.bind(this)} value={this.state.value}/>
            }
        }

        const component = mount(<Demo/>);
        component.find('input').simulate('change', {target: {value: '111'}});
        expect(component.instance().state.value).toEqual('111');
        expect(component.find('input').prop('value')).toBe('111');
    });
});