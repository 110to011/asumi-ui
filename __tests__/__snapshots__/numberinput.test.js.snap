// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput create component 1`] = `
<NumberInput
  dataFormat={[Function]}
  defaultValue=""
  max={9007199254740991}
  min={-9007199254740991}
  step={1}
>
  <Input
    append={
      <span
        className="el-number-input-handler"
      >
        <span
          className="fa fa-angle-up el-number-input-up"
          onClick={[Function]}
        />
        <span
          className="fa fa-angle-down el-number-input-down"
          onClick={[Function]}
        />
      </span>
    }
    className="el-number-input"
    max={9007199254740991}
    min={-9007199254740991}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  >
    <div
      className="el-input-wrapper el-number-input"
    >
      <div
        className="el-input el-number-input"
      >
        <input
          max={9007199254740991}
          min={-9007199254740991}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <span
        className="el-input-append"
      >
        <span
          className="el-number-input-handler"
        >
          <span
            className="fa fa-angle-up el-number-input-up"
            onClick={[Function]}
          />
          <span
            className="fa fa-angle-down el-number-input-down"
            onClick={[Function]}
          />
        </span>
      </span>
    </div>
  </Input>
</NumberInput>
`;

exports[`NumberInput define disabled 1`] = `
<NumberInput
  dataFormat={[Function]}
  defaultValue=""
  disabled={true}
  max={9007199254740991}
  min={-9007199254740991}
  step={1}
>
  <Input
    append={
      <span
        className="el-number-input-handler"
      >
        <span
          className="fa fa-angle-up el-number-input-up"
          onClick={[Function]}
        />
        <span
          className="fa fa-angle-down el-number-input-down"
          onClick={[Function]}
        />
      </span>
    }
    className="el-number-input el-number-input-disabled"
    disabled={true}
    max={9007199254740991}
    min={-9007199254740991}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  >
    <div
      className="el-input-wrapper el-number-input el-number-input-disabled"
    >
      <div
        className="el-input el-number-input el-number-input-disabled"
      >
        <input
          disabled={true}
          max={9007199254740991}
          min={-9007199254740991}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <span
        className="el-input-append"
      >
        <span
          className="el-number-input-handler"
        >
          <span
            className="fa fa-angle-up el-number-input-up"
            onClick={[Function]}
          />
          <span
            className="fa fa-angle-down el-number-input-down"
            onClick={[Function]}
          />
        </span>
      </span>
    </div>
  </Input>
</NumberInput>
`;

exports[`NumberInput define size[large] 1`] = `
<NumberInput
  dataFormat={[Function]}
  defaultValue=""
  max={9007199254740991}
  min={-9007199254740991}
  size="large"
  step={1}
>
  <Input
    append={
      <span
        className="el-number-input-handler"
      >
        <span
          className="fa fa-angle-up el-number-input-up"
          onClick={[Function]}
        />
        <span
          className="fa fa-angle-down el-number-input-down"
          onClick={[Function]}
        />
      </span>
    }
    className="el-number-input"
    max={9007199254740991}
    min={-9007199254740991}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    size="large"
    type="text"
    value=""
  >
    <div
      className="el-input-wrapper el-number-input el-large"
    >
      <div
        className="el-input el-number-input el-large"
      >
        <input
          max={9007199254740991}
          min={-9007199254740991}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <span
        className="el-input-append"
      >
        <span
          className="el-number-input-handler"
        >
          <span
            className="fa fa-angle-up el-number-input-up"
            onClick={[Function]}
          />
          <span
            className="fa fa-angle-down el-number-input-down"
            onClick={[Function]}
          />
        </span>
      </span>
    </div>
  </Input>
</NumberInput>
`;

exports[`NumberInput define size[small] 1`] = `
<NumberInput
  dataFormat={[Function]}
  defaultValue=""
  max={9007199254740991}
  min={-9007199254740991}
  size="small"
  step={1}
>
  <Input
    append={
      <span
        className="el-number-input-handler"
      >
        <span
          className="fa fa-angle-up el-number-input-up"
          onClick={[Function]}
        />
        <span
          className="fa fa-angle-down el-number-input-down"
          onClick={[Function]}
        />
      </span>
    }
    className="el-number-input"
    max={9007199254740991}
    min={-9007199254740991}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    size="small"
    type="text"
    value=""
  >
    <div
      className="el-input-wrapper el-number-input el-small"
    >
      <div
        className="el-input el-number-input el-small"
      >
        <input
          max={9007199254740991}
          min={-9007199254740991}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyPress={[Function]}
          type="text"
          value=""
        />
      </div>
      <span
        className="el-input-append"
      >
        <span
          className="el-number-input-handler"
        >
          <span
            className="fa fa-angle-up el-number-input-up"
            onClick={[Function]}
          />
          <span
            className="fa fa-angle-down el-number-input-down"
            onClick={[Function]}
          />
        </span>
      </span>
    </div>
  </Input>
</NumberInput>
`;
